cpp/src/gandiva/lru_cache_test.cc

```diff
@@ -40,25 +40,25 @@ class TestLruCache : public ::testing::Test {
   TestLruCache() : cache_(2) {}
 
  protected:
-  LruCache<TestCacheKey, std::string> cache_;
+  LruCache<TestLruCacheKey, std::string> cache_;
 };
 
 TEST_F(TestLruCache, TestEvict) {
-  cache_.insert(TestCacheKey(1), "hello");
-  cache_.insert(TestCacheKey(2), "hello");
-  cache_.insert(TestCacheKey(1), "hello");
-  cache_.insert(TestCacheKey(3), "hello");
+  cache_.insert(TestLruCacheKey(1), ValueCacheObject<std::string>("hello", 0));
+  cache_.insert(TestLruCacheKey(2), ValueCacheObject<std::string>("hello", 0));
+  cache_.insert(TestLruCacheKey(1), ValueCacheObject<std::string>("hello", 0));
+  cache_.insert(TestLruCacheKey(3), ValueCacheObject<std::string>("hello", 0));
   // should have evicted key 1
   ASSERT_EQ(2, cache_.size());
-  ASSERT_EQ(cache_.get(TestCacheKey(1)), arrow::util::nullopt);
+  ASSERT_EQ(cache_.get(TestLruCacheKey(1)), arrow::util::nullopt);
 }
 
 TEST_F(TestLruCache, TestLruBehavior) {
-  cache_.insert(TestCacheKey(1), "hello");
-  cache_.insert(TestCacheKey(2), "hello");
-  cache_.get(TestCacheKey(1));
-  cache_.insert(TestCacheKey(3), "hello");
+  cache_.insert(TestLruCacheKey(1), ValueCacheObject<std::string>("hello", 0));
```

Comment:
Here is the same consideration of varying the string to be inserted into the cache.